lib/cli/cli.js

```diff
@@ -118,139 +118,142 @@ class CLI {
-              `Node ${process.version} is not tested against Ember CLI on your platform.${recommendation}`
-            );
-          }
-        }
+      getOptionArgs('--verbose', commandArgs).forEach(arg => {
+        process.env[`EMBER_VERBOSE_${arg.toUpperCase()}`] = 'true';
+      });
 
-        platformCheckerToken.stop();
+      let platformCheckerToken = heimdall.start('platform-checker');
 
-        logger.info('command: %s', commandName);
+      const PlatformChecker = require('../utilities/platform-checker');
+      let platform = new PlatformChecker(process.version);
+      let recommendation =
+        ' We recommend that you use the most-recent "Active LTS" version of Node.js.' +
+        ' See https://git.io/v7S5n for details.';
 
-        if (!this.testing) {
-          process.chdir(environment.project.root);
-          let skipInstallationCheck = commandArgs.indexOf('--skip-installation-check') !== -1;
-          if (environment.project.isEmberCLIProject() && !skipInstallationCheck) {
-            const InstallationChecker = require('../models/installation-checker');
-            new InstallationChecker({ project: environment.project }).checkInstallations();
-          }
+      if (!this.testing) {
+        if (platform.isDeprecated) {
+          this.ui.writeDeprecateLine(`Node ${process.version} is no longer supported by Ember CLI.${recommendation}`);
         }
 
-        let instrumentation = this.instrumentation;
-        let onCommandInterrupt;
+        if (!platform.isTested) {
+          this.ui.writeWarnLine(
+            `Node ${process.version} is not tested against Ember CLI on your platform.${recommendation}`
+          );
+        }
+      }
 
-        let runPromise = Promise.resolve()
-          .then(() => {
-            instrumentation.stopAndReport('init');
-            instrumentation.start('command');
+      platformCheckerToken.stop();
 
-            loggerTesting.info('cli: command.beforeRun');
-            onProcessInterrupt.addHandler(onCommandInterrupt);
+      logger.info('command: %s', commandName);
 
-            return command.beforeRun(commandArgs);
-          })
-          .then(() => {
-            loggerTesting.info('cli: command.validateAndRun');
+      if (!this.testing) {
+        process.chdir(environment.project.root);
+        let skipInstallationCheck = commandArgs.indexOf('--skip-installation-check') !== -1;
+        if (environment.project.isEmberCLIProject() && !skipInstallationCheck) {
+          const InstallationChecker = require('../models/installation-checker');
+          new InstallationChecker({ project: environment.project }).checkInstallations();
+        }
+      }
 
-            return command.validateAndRun(commandArgs);
-          })
-          .then(result => {
-            instrumentation.stopAndReport('command', commandName, commandArgs);
+      let instrumentation = this.instrumentation;
+      let onCommandInterrupt;
+
+      let runPromise = Promise.resolve()
```

Comment:
This is left out of the async/await refactor specifically because it is intertwined with the `onCommandInterrupt` function in a way that is not trivially possible with `async` / `await`.

Reply:
Let's created a `async () => {}` here, so we can coordinate on `onCommandInterrupt` and `runPromise` as currently implemented, but using cleaner more maintainable syntax

Reply:
Ya, great suggestion! Will update.

Reply:
Updated with that...